### PR TITLE
add tests and fixed bugs in homework.py

### PR DESCRIPTION
**Pull request by oleksii-manzik**

### CODE REVIEW

File: tests/tests.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    def test_sum_of_corners_calc(self):
+        for i in range(1, 10):
+            with self.subTest(name=f'sum of {i} corners'):
+                if i <= 4:
+                    self.assertEqual(
+                        self.test_rectangle_1.get_sum_of_corners(i),
+                        i * 90)
+                else:
+                    with self.assertRaises(ValueError):
+                        self.test_rectangle_1.get_sum_of_corners(i)
```
Comment (alchemixXx) [inline tests/tests.py]: split it into two separate tests: with good values and  bad values. It will be more readable

File: tests/tests.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        with self.subTest(name='test_rectangle_1'):
+            with self.assertRaises(ValueError):
+                self.test_rectangle_1.get_radius_of_inscribed_circle()
+
+        with self.subTest(name='test_rectangle_2'):
+            with self.assertRaises(ValueError):
+                self.test_rectangle_2.get_radius_of_inscribed_circle()
+
+        with self.subTest(name='test_rectangle_3'):
+            self.assertEqual(
```
Comment (alchemixXx) [inline tests/tests.py]: it is really necessary to make 3 subtests? Or just one with 3 asserts will be more clear?

**Review by alchemixXx — CHANGES_REQUESTED** [review]

File: tests/tests.py (hunk 1 above)
Comment (oleksii-manzik) [reply tests/tests.py]: fixed

File: tests/tests.py (hunk 2 above)
Comment (oleksii-manzik) [reply tests/tests.py]: fixed

File: tests/tests.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        self.assertEqual(self.test_rectangle_3.get_rectangle_perimeter(), 20)
+
+    def test_rectangle_square_calc(self):
+        self.assertEqual(self.test_rectangle_1.get_rectangle_square(), 4)
+        self.assertEqual(self.test_rectangle_2.get_rectangle_square(), 30)
+        self.assertEqual(self.test_rectangle_3.get_rectangle_square(), 25)
+
+    def test_sum_of_corners_calc(self):
+        for i in range(1, 5):
+            with self.subTest(name=f'sum of {i} corners'):
```
Comment (alchemixXx) [inline tests/tests.py]: do you still need subtests here?

File: tests/tests.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                self.assertEqual(
+                    self.test_rectangle_1.get_sum_of_corners(i), i * 90)
+
+    def test_sum_of_corners_error_raise(self):
+        for i in range(5, 10):
+            with self.subTest(name=f'sum of {i} corners'):
```
Comment (alchemixXx) [inline tests/tests.py]: do you still need subtests here?

**Review by alchemixXx — DISMISSED** [review]

File: tests/tests.py (hunk 3 above)
Comment (oleksii-manzik) [reply tests/tests.py]: idk it looks cool in pycharm :)

image

Comment (oleksii-manzik) [reply tests/tests.py]: but okay I will remove subtests. thanks

**Pull request merged**